backend/src/main/kotlin/app/ehrenamtskarte/backend/cards/webservice/schema/CardMutationService.kt

```diff
@@ -115,6 +124,88 @@ class CardMutationService {
         }
     }
 
+    @GraphQLDescription("Creates a new digital entitlementcard from self-service portal")
+    fun createCardFromSelfService(
+        dfe: DataFetchingEnvironment,
+        project: String,
+        encodedCardInfo: String,
+        generateStaticCode: Boolean
+    ): CardCreationResultModel {
+        // TODO disable for production until the Koblenz project is ready to go live
+        val context = dfe.getContext<GraphQLContext>()
+        if (context.backendConfiguration.production) {
+            throw ForbiddenException()
+        }
+        if (project != KOBLENZ_PASS_PROJECT) {
+            throw NotKoblenzProjectException()
+        }
+
+        val cardInfo = parseEncodedCardInfo(encodedCardInfo)
+        val regionId = cardInfo.extensions.extensionRegion.regionId
+        transaction { RegionsRepository.findByIdInProject(project, regionId) } ?: throw RegionNotFoundException()
+
+        val user = KoblenzUser(
+            cardInfo.fullName,
+            cardInfo.extensions.extensionBirthday.birthday,
+            cardInfo.extensions.extensionKoblenzReferenceNumber.referenceNumber
+        )
+
+        val userHash = Argon2IdHasher.hashKoblenzUserData(user)
+
+        val userEntitlements = transaction { UserEntitlementsRepository.findUserEntitlements(userHash.toByteArray(), regionId) }
+        if (userEntitlements == null || userEntitlements.revoked || userEntitlements.endDate.isBefore(LocalDate.now())) {
+            throw InvalidUserEntitlementsException()
+        }
+
+        val cardInfoWithExpDayAndStartDay = enrichCardInfoWithExpDayAndStartDay(
```

Comment:
please also enrich regionId
(optional) i don't like shortening function names since its not that easy to understand so i would prefer
`enrichCardInfoWithRegionIdExpirationDayAndStartDay`
Or just name it `enrichMissingCardInfo` if it gets too long

Reply:
Renamed to enrichCardInfo(), as basically from the implementation perspective it doesn't matter if the data is missing or not, it will be added anyway